### PR TITLE
Better OSS RoP action

### LINKED ISSUE #604: GitHub action for OSS RoP unnecessarily updates .gitignore
https://github.com/SAP/fosstars-rating-core/blob/80d7936ab38a572056e463dbdf6e2dc6f00c69f3/.github/actions/oss-rules-of-play-action/entrypoint.sh#L59

The GitHub action for OSS RoP updates `.gitignore` file when it runs. As a result, the file unnecessarily grows.

### LINKED ISSUE #605: Check if a token is available in the GitHub action for OSS RoP
https://github.com/SAP/fosstars-rating-core/blob/80d7936ab38a572056e463dbdf6e2dc6f00c69f3/.github/actions/oss-rules-of-play-action/entrypoint.sh#L49

The action expects a token but there is no check for that.

### PR DESCRIPTION
**Pull request by artem-smotrakov**
Closes #604
Closes #605

### CODE REVIEW

File: .github/actions/oss-rules-of-play-action/entrypoint.sh [hunk 1]
```diff
@@ -15,6 +15,13 @@ if [ "$FOSSTARS_VERSION" = "" ]; then
     exit 1
 fi
 
+if [ "$TOKEN" = "" ]; then
+    echo "Achtung! No token provided or it is empty!"
+    TOKEN_OPTION=""
```
Comment (sourabhsparkala) [inline .github/actions/oss-rules-of-play-action/entrypoint.sh:20]: Do we exit if there is no token?

**Review by sourabhsparkala — APPROVED** [review]
A minor comment

File: .github/actions/oss-rules-of-play-action/entrypoint.sh (hunk 1 above)
Comment (artem-smotrakov) [reply .github/actions/oss-rules-of-play-action/entrypoint.sh:20]: No, the action will continue to run but without token.

File: .github/actions/oss-rules-of-play-action/entrypoint.sh [hunk 2]
```diff
@@ -55,9 +62,11 @@ if [ $? -ne 0 ]; then
     exit 1
 fi
 
+# Cleanup
+rm -rf fosstars-rating-core > /dev/null 2>&1
+rm -rf .fosstars > /dev/null 2>&1
+
```
Comment (sourabhsparkala) [inline .github/actions/oss-rules-of-play-action/entrypoint.sh:68]: Could this be affected by some folder permissions?

Comment (artem-smotrakov) [reply .github/actions/oss-rules-of-play-action/entrypoint.sh:68]: Those directories are created by the action (by same process/user that tries to delete them), so there should not be issues with permissions.

**Pull request merged**